src/main/java/org/jenkinsci/plugins/workflow/multibranch/BranchJobProperty.java

```diff
@@ -67,23 +66,25 @@ synchronized void setBranch(@Nonnull Branch branch) {
             @Override public boolean hasPermission(Authentication a, Permission permission) {
                 // This project is managed by its parent and may not be directly configured or deleted by users.
                 // Note that Item.EXTENDED_READ may still be granted, so you can still see Snippet Generator, etc.
-                if (branch instanceof Branch.Dead && (permission == Job.BUILD)) {
-                    // if dead, nobody, not even SYSTEM may build this job
-                    return false;
-                } else if (ACL.SYSTEM.equals(a)) {
-                    return true; // e.g., DefaultDeadBranchStrategy.runDeadBranchCleanup
```

Comment:
As noted in my comment in #45, this clause (now actually referring to `ComputedFolder.updateChildren`) is obsolete due to the exclusion of `Branch.Dead` below. The `deletedBranch` test confirms it.